framework/wazuh/core/mitre.py

```diff
@@ -0,0 +1,548 @@
+                 count: bool = True, table: str = '', sort: dict = None, default_sort_field: str = 'id',
+                 default_sort_order='ASC', fields=None, search: dict = None, select: list = None,
+                 min_select_fields=None, filters=None, request_slice=500):
+        """Create an instance of WazuhDBQueryMitre query."""
+
+        if filters is None:
+            filters = {}
+
+        WazuhDBQuery.__init__(self, offset=offset, limit=limit, table=table, sort=sort, search=search, select=select,
+                              fields=fields, default_sort_field=default_sort_field,
+                              default_sort_order=default_sort_order, filters=filters, query=query, count=count,
+                              get_data=True, min_select_fields=min_select_fields,
+                              backend=WazuhDBBackend(query_format='mitre', request_slice=request_slice))
+
+        self.relation_fields = set()  # This variable contains valid fields not included in the database (relations)
+
+    def _filter_status(self, status_filter):
+        pass
+
+
+class WazuhDBQueryMitreMetadata(WazuhDBQueryMitre):
+
+    def __init__(self):
+        """Create an instance of WazuhDBQueryMitreMetadata query."""
+
+        min_select_fields = {'key', 'value'}
+        fields = {'key': 'key', 'value': 'value'}
+
+        WazuhDBQueryMitre.__init__(self, table='metadata', min_select_fields=min_select_fields, fields=fields,
+                                   default_sort_field='key')
+
+    def _filter_status(self, status_filter):
+        pass
+
+
+class WazuhDBQueryMitreRelational(WazuhDBQueryMitre):
+
+    def __init__(self, table: str = None, offset: int = 0, limit: Union[int, None] = common.database_limit,
+                 query: str = '', count: bool = True, sort: dict = None, default_sort_order: str = 'ASC',
+                 default_sort_field: str = None, fields=None, search: dict = None, select: list = None,
+                 min_select_fields=None, filters=None, dict_key: str = None, request_slice=RELATIONAL_REQUEST_SLICE):
+        """WazuhDBQueryMitreRelational constructor
+        This class will always generate dictionaries with two keys, this is because it handles relational tables,
+        where the relationship of two objects is specified.
+
+        Parameters
+        ----------
+        dict_key : str
+            The value of this key will be the key of the output dictionary.
+            The value of the output dictionary will be the value of the remaining key.
+        """
+
+        if filters is None:
+            filters = {}
+        if min_select_fields is None:
+            if table == 'phase':
+                self.min_select_fields = {'tactic_id', 'tech_id'}
+                default_sort_field = 'tactic_id'
+            elif table == 'mitigate' or table == 'use':
+                # source_id = mitigation_id or group_id or software_id, target_id = technique_id
```

Comment:
?

Reply:
updated comment,
source_id refers to mitigation_id, group_id or software_id
target_id alwauys refers to technique_id